server/container/guice/cassandra-rabbitmq-guice/src/test/java/org/apache/james/modules/blobstore/BlobStoreModulesChooserTest.java

```diff
@@ -0,0 +1,105 @@
+
+import org.apache.commons.configuration2.PropertiesConfiguration;
+import org.apache.james.FakePropertiesProvider;
+import org.apache.james.blob.union.HybridBlobStore;
+import org.apache.james.modules.mailbox.ConfigurationComponent;
+import org.junit.jupiter.api.Test;
+
+class BlobStoreModulesChooserTest {
+    @Test
+    void providesHybridBlobStoreConfigurationShouldThrowWhenNegative() {
+        HybridDeclarationModule module = new HybridDeclarationModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("hybrid.size.threshold", -1);
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(ConfigurationComponent.NAME, configuration)
+            .build();
+
+        assertThatThrownBy(() -> module.providesHybridBlobStoreConfiguration(propertyProvider))
+            .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void providesHybridBlobStoreConfigurationShouldNotThrowWhenZero() {
+        HybridDeclarationModule module = new HybridDeclarationModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("hybrid.size.threshold", 0);
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(ConfigurationComponent.NAME, configuration)
+            .build();
+
+        assertThat(module.providesHybridBlobStoreConfiguration(propertyProvider))
+            .isEqualTo(new HybridBlobStore.Configuration(0));
+    }
+
+    @Test
+    void providesHybridBlobStoreConfigurationShouldReturnConfiguration() {
+        HybridDeclarationModule module = new HybridDeclarationModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("hybrid.size.threshold", 36);
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(ConfigurationComponent.NAME, configuration)
+            .build();
+
+        assertThat(module.providesHybridBlobStoreConfiguration(propertyProvider))
+            .isEqualTo(new HybridBlobStore.Configuration(36));
+    }
+
+    @Test
+    void providesHybridBlobStoreConfigurationShouldReturnConfigurationWhenLegacyFile() {
+        HybridDeclarationModule module = new HybridDeclarationModule();
+        PropertiesConfiguration configuration = new PropertiesConfiguration();
+        configuration.addProperty("hybrid.size.threshold", 36);
+        FakePropertiesProvider propertyProvider = FakePropertiesProvider.builder()
+            .register(ConfigurationComponent.LEGACY, configuration)
+            .build();
+
+        assertThat(module.providesHybridBlobStoreConfiguration(propertyProvider))
+            .isEqualTo(new HybridBlobStore.Configuration(36));
+    }
+
+    @Test
+    void provideBlobStoreShouldReturnCassandraBlobStoreWhenCassandraConfigured() {
```

Comment:
Names of tests and test themselves (regarding this one and the one below) are mixed up.

This one is about `ObjectStorageBlobStore`, and the one below about `CassandraBlobStore`

Reply:
Nice catch! Not my code but definitly nice to fix